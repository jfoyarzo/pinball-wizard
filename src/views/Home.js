import { React, useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { Row, Col, Container } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { getMachines } from '../redux/machines/machinesSlice';
import usa from '../img/usa_map.png';
import pinballIcon from '../img/pinball_icon.png';

const Home = () => {
  const dispatch = useDispatch();
  useEffect(() => {
    dispatch(getMachines());
  }, [dispatch]);

  const totalMachines = useSelector((state) => state.totalMachines);
  const top = useSelector((state) => state.topCities);

  return (
    <div className="home-wrapper">
      <Row xs="2" style={{ height: '25vh', padding: '1em', backgroundColor: 'blue' }}>
        <Col style={{
          backgroundImage: `url(${usa})`, backgroundColor: 'blue', backgroundBlendMode: 'multiply', backgroundSize: '20vh', backgroundRepeat: 'no-repeat', backgroundPosition: 'center',
        }}
        />
        <Col>
          <h2>USA</h2>
          <p>
            Machine count:
            {' '}
            {totalMachines}
          </p>
        </Col>
      </Row>
      <Container className="border border-primary">
        <p>Top Cities</p>
      </Container>
      <div className="top-container">
        <Row>
          {top.map((region) => (
            <Col
              xs="6"
              key={region.city}
              className="p-4 border"
              style={{
                backgroundImage: `url(${pinballIcon})`, backgroundSize: '13vh', backgroundRepeat: 'no-repeat', backgroundPosition: 'top', backgroundColor: 'blue', backgroundBlendMode: 'multiply',
              }}
            >
              <Link to={`/${region.city}`}>
                <div className="d-flex flex-column justify-content-end" style={{ height: '15vh' }}>
                  <h3>{region.city}</h3>
                  <p>
                    Machine Count:
                    {' '}
                    {region.machineCount}
                  </p>
                </div>
              </Link>
            </Col>
          ))}
        </Row>
      </div>
    </div>
  );
};

export default Home;
