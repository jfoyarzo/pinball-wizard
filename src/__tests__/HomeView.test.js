import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/configureStore';
import Home from '../views/Home';

describe('Home view tests', () => {
  it('renders correctly', () => {
    const home = render(
      <Router>
        <Provider store={store}>
          <Home />
        </Provider>
      </Router>,
    );
    expect(home).toMatchSnapshot();
  });

  it('displays the correct country', () => {
    const { getByText } = render(
      <Router>
        <Provider store={store}>
          <Home />
        </Provider>
      </Router>,
    );
    expect(getByText('USA')).toBeInTheDocument();
  });
});
