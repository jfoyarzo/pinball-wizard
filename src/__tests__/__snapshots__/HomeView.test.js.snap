// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home view tests renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="home-wrapper container-fluid p-0"
      >
        <div
          class="row"
          style="height: 22vh; padding: 1em; background-color: rgb(88, 137, 228);"
        >
          <div
            class="col-7"
            style="background-image: url(usa_map.png); background-color: rgb(88, 137, 228); background-blend-mode: multiply; background-size: 25vh; background-repeat: no-repeat;"
          />
          <div
            class="d-flex flex-column justify-content-center ps-4 col"
          >
            <h1
              class="mb-0"
            >
              USA
            </h1>
            <p>
               
              machines
            </p>
          </div>
        </div>
        <div
          class="container"
          style="background-color: rgb(53, 84, 139);"
        >
          <p
            class="fw-bold"
            style="font-size: 1em;"
          >
            MACHINES BY CITY
          </p>
        </div>
        <div
          class="top-container container-fluid"
        >
          <div
            class="cities-row row"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="home-wrapper container-fluid p-0"
    >
      <div
        class="row"
        style="height: 22vh; padding: 1em; background-color: rgb(88, 137, 228);"
      >
        <div
          class="col-7"
          style="background-image: url(usa_map.png); background-color: rgb(88, 137, 228); background-blend-mode: multiply; background-size: 25vh; background-repeat: no-repeat;"
        />
        <div
          class="d-flex flex-column justify-content-center ps-4 col"
        >
          <h1
            class="mb-0"
          >
            USA
          </h1>
          <p>
             
            machines
          </p>
        </div>
      </div>
      <div
        class="container"
        style="background-color: rgb(53, 84, 139);"
      >
        <p
          class="fw-bold"
          style="font-size: 1em;"
        >
          MACHINES BY CITY
        </p>
      </div>
      <div
        class="top-container container-fluid"
      >
        <div
          class="cities-row row"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
